pdx-extract/app/logger.py

```diff
@@ -0,0 +1,32 @@
+import logging
+from datetime import datetime
+
+from pythonjsonlogger import jsonlogger
+
+from app import config
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def add_fields(self, log_record, record, message_dict):
+        super(CustomJsonFormatter, self).add_fields(log_record, record, message_dict)
+        if not log_record.get("timestamp"):
+            now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+            log_record["timestamp"] = now
+        if log_record.get("level"):
+            log_record["level"] = log_record["level"].upper()
+        else:
+            log_record["level"] = record.levelname
+
+
+modules = ["requests", "selenium", "urllib3", "pyppeteer", "websockets", "asyncio"]
+
+for module in modules:
+    logging.getLogger(module).setLevel(logging.WARNING)
```

Comment:
This pretty tightly couples to logger to the modules it supports.
In this scheme, the logger configures the modules, rather than the modules configuring the logger. Consider exposing a default logger level and having each module configure their own logger in construction (or, create a common HasLogging superclass and leverage Python's multiple inheritance to implement a constructor for that class which initializes the logger appropriately)

Reply:
That's a really interesting point, that shows another violation of the open/closed principle.

So you're suggesting setting the default logging level to WARNING and a granular per-module logging level -- which can be whatever is defined in the configuration file.

I'll have to think how to implement that in a centralized way, that prevents me from having to declare the logging level on each module. That superclass suggestion is something I'll have to think about, I can't see it ATM.